### PR TITLE
Fix "Auto-hide item methods documentation" setting

### LINKED ISSUE #58825: The "Auto-hide item methods' documentation" option doesn't work

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #58825.

r? @QuietMisdreavus

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -2374,6 +2355,30 @@ if (!DOMTokenList.prototype.remove) {
         };
     });
 
+    // In the search display, allows to switch between tabs.
+    function printTab(nb) {
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/main.js]: Why was this function moved? It doesn't seem to have anything to do with this PR, and it breaks the ability to do `git blame` analysis on it. (And probably will create a merge conflict with https://github.com/rust-lang/rust/pull/58364 on top of that.)

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: It was between functions that should be along each other. I can move it back if you want?

Comment (QuietMisdreavus) [reply src/librustdoc/html/static/main.js]: It's fine; i won't let that hold up the PR.

**Pull request merged**